tests/integration/security/util/framework.go

```diff
@@ -136,6 +148,48 @@ func SetupApps(ctx resource.Context, i istio.Instance, apps *EchoDeployments, bu
 		builder.With(nil, EchoConfig(HeadlessNakedSvc, apps.Namespace1, true, echo.NewAnnotations().
 			SetBool(echo.SidecarInject, false), c[0]))
 	}
+	newEchoConfigForTestPassThroughFilterChain := func(service string, ns namespace.Instance, cluster resource.Cluster) echo.Config {
+		return echo.Config{
+			Service:   service,
+			Namespace: ns,
+			Subsets:   []echo.SubsetConfig{{}},
+			Ports: []echo.Port{
+				{
+					Name:     "grpc",
+					Protocol: protocol.GRPC,
+				},
+			},
+			Cluster: cluster,
+			// The port 8085,8086,8087,8088 will be defined only in the workload and not in the k8s service.
+			WorkloadOnlyPorts: []echo.WorkloadPort{
+				{
+					Port:     8085,
+					Protocol: protocol.HTTP,
+				},
+				{
+					Port:     8086,
+					Protocol: protocol.HTTP,
+				},
+				{
+					Port:     8087,
+					Protocol: protocol.TCP,
+				},
+				{
+					Port:     8088,
+					Protocol: protocol.TCP,
+				},
+			},
+		}
+	}
+	for _, cluster := range ctx.Clusters() {
+		builder.
+			With(nil, newEchoConfigForTestPassThroughFilterChain(ASvc, apps.Namespace2, cluster)).
```

Comment:
This will overwrite Deployment in kube for A, B, C, D. The main difference I see is the WorkloadOnly ports. Can we just include those in the `EchoConfig` fn?

Reply:
Hi Steven, I have address your comment in this PR. https://github.com/istio/istio/pull/28720